flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -313,11 +296,15 @@ private <T> Optional<T> getLatestPriorProperty(Function<ArchivedExecution, T> ex
      * @return The latest prior execution location, or null, if there is none, yet.
      */
     public Optional<TaskManagerLocation> findLatestPriorLocation() {
```

Comment:
Can we rephrase these `findLatestPrior*` and `setLatestPrior*` methods. Because we're setting it now when assigning the resource it means that we actually return the latest assignment and not the latest prior assignment...

Reply:
Had the same thoughts, but one could argue that whether it is set or find through some fancy algorithm is an implementation detail, so I reverted it.

Reply:
it's not about the set and find but the `LatestPrior` substring that is not valid anymore

Reply:
ah